libs/search/src/lib/state/effects.ts

```diff
@@ -55,53 +56,67 @@ export class SearchEffects {
         SET_PAGINATION,
         PAGINATE
       ),
-      switchMap(() => of(new ClearResults(), new RequestMoreResults()))
+      switchMap((action: SearchActions) =>
+        of(new ClearResults(action.id), new RequestMoreResults(action.id))
+      )
     )
   )
 
   scroll$ = createEffect(() =>
     this.actions$.pipe(
       ofType(SCROLL),
-      map(() => new RequestMoreResults())
+      map((action: SearchActions) => new RequestMoreResults(action.id))
     )
   )
 
   loadResults$ = createEffect(() =>
     this.actions$.pipe(
       ofType(REQUEST_MORE_RESULTS),
-      switchMap(() => this.authService.authReady()), // wait for auth to be known
-      withLatestFrom(this.store$.pipe(select(getSearchStateSearch))),
-      switchMap(([_, state]) =>
-        this.searchService.search(
-          'bucket',
-          JSON.stringify(
-            this.esService.search(state, ElasticsearchMetadataModels.SUMMARY)
-          )
-        )
-      ),
-      switchMap((response: SearchResponse<any>) => {
-        const records = this.esMapper.toRecordSummary(
-          response,
-          this.searchService.configuration.basePath
+      switchMap((action: SearchActions) =>
+        this.authService.authReady().pipe(
```

Comment:
Is it intentional to pipe all the work inside the authReady stream ? Why not go back to the actions$ stream like before ?